File: frc2018_5/src/main/java/ca/warp7/frc2018_5/subsystems/Navx.java
package ca.warp7.frc2018_5.subsystems;

import ca.warp7.frc.core.IComponent;
import com.kauailabs.navx.frc.AHRS;
import edu.wpi.first.wpilibj.SPI;

public class Navx implements IComponent {
    private AHRS ahrs;

    @Override
    public void onConstruct() {
        ahrs = new AHRS(SPI.Port.kMXP);
    }

    AHRS getAhrs() {
        return ahrs;
    }
}

File: frc2018_5/src/main/java/ca/warp7/frc2018_5/input/NavX.kt
package ca.warp7.frc2018_5.input

import ca.warp7.frc.next.InputSystem
import com.kauailabs.navx.frc.AHRS
import edu.wpi.first.wpilibj.SPI

object NavX : InputSystem {
    private val mAHRS = AHRS(SPI.Port.kMXP)

    var yaw = 0.0
    var calibrated = false

    override fun onMeasure() {
        if (!calibrated && !mAHRS.isCalibrating) {
            calibrated = true
        }
        yaw = Math.toRadians(mAHRS.fusedHeading.toDouble())
    }
}